nixos/modules/programs/appimage.nix

```diff
@@ -18,16 +23,31 @@ in
   };
 
   config = lib.mkIf cfg.enable {
-    boot.binfmt.registrations.appimage = lib.mkIf cfg.binfmt {
-      wrapInterpreterInShell = false;
-      interpreter = lib.getExe cfg.package;
-      recognitionType = "magic";
-      offset = 0;
-      mask = ''\xff\xff\xff\xff\x00\x00\x00\x00\xff\xff\xff'';
-      magicOrExtension = ''\x7fELF....AI\x02'';
-    };
+    boot.binfmt.registrations.appimage = lib.mkIf cfg.binfmt (
+      let
+        appimage_common = {
+          wrapInterpreterInShell = false;
+          interpreter = lib.getExe cfg.package;
+          recognitionType = "magic";
+          offset = 0;
+          mask = ''\xff\xff\xff\xff\x00\x00\x00\x00\xff\xff\xff'';
+        };
+      in
+      {
+        appimage_type_1 = appimage_common // {
+          magicOrExtension = ''\x7fELF....AI\x01'';
```

Comment:
I don't know the mechanism here well enough but is it perhaps possible to just ignore the lowest byte here?

Reply:
Not sure about that, but I'm referencing appimagelauncher (which I'm also trying to package): https://github.com/TheAssassin/AppImageLauncher/blob/master/resources/binfmt.d/appimagelauncher.conf.in